src/gluonnlp/utils/misc.py

```diff
@@ -363,8 +363,11 @@ def download(url: str,
     """
     is_s3 = url.startswith(S3_PREFIX)
     if is_s3:
-        boto3 = try_import_boto3()
+        boto3, botocore = try_import_boto3()
         s3 = boto3.resource('s3')
+        if boto3.session.Session().get_credentials() is None:
+            from botocore.handlers import disable_signing
+            s3.meta.client.meta.events.register('choose-signer.s3.*', disable_signing)
```

Comment:
does this behavior need be parametrized in the download function?

Reply:
What's your suggestion? The `s3` resource is created only in the download function

Reply:
I was thinking about having a `use_unsigned` argument so that anonymous downloading is explicit. Otherwise user may not be aware if the credential passing failed unexpectedly.

Reply:
I think we're just working around a bug in boto, thus there is no value in exposing it to users. There is no reason for boto to fail to download public files if no credentials are passed.